app/src/main/java/de/westnordost/streetcomplete/settings/SettingsFragment.kt

```diff
@@ -137,6 +205,10 @@ class SettingsFragment : PreferenceFragmentCompat(), HasTitle,
                 AppCompatDelegate.setDefaultNightMode(theme.appCompatNightMode)
                 activity?.recreate()
             }
+            Prefs.LANGUAGE_SELECT -> {
+                val intent = Intent(context?.applicationContext, MainActivity::class.java)
+                this.startActivity(intent)
```

Comment:
I think what you want to do is
```suggestion
                activity?.recreate()
```

Reply:
That seemed to switch the language only for the settings activity but not for the rest of the app.
Anyways, we don't need to switch it on the fly and it would be okay prompt the user for a full restart of the app, so this could be avoided.

Reply:
Restarting the app would be fine, maybe it could be done automatically

Reply:
As a user, I wouldn't want the app to restart as soon as I picked a language, because that would boot me out of the settings. I'd be happy if the app restarted as soon as I closed the settings screen.